src/Orc.FilterBuilder/Orc.FilterBuilder.Shared/Extensions/ConditionsLinqExtentions.cs

```diff
@@ -0,0 +1,270 @@
+                case Condition.GreaterThan:
+                    return Expression.GreaterThan(Expression.Call(
+                            typeof(string).GetMethod("Compare", new[] { typeof(string), typeof(string), typeof(StringComparison) }),
+                            propertyExpression, valueExpression, Expression.Constant(StringComparison.InvariantCultureIgnoreCase)),
+                        Expression.Constant(0));
+                case Condition.GreaterThanOrEqualTo:
+                    return Expression.GreaterThanOrEqual(
+                        Expression.Call(
+                            typeof(string).GetMethod("Compare", new[] { typeof(string), typeof(string), typeof(StringComparison) }),
+                            propertyExpression, valueExpression, Expression.Constant(StringComparison.InvariantCultureIgnoreCase)), Expression.Constant(0));
+                case Condition.LessThan:
+                    return Expression.LessThan(Expression.Call(
+                            typeof(string).GetMethod("Compare", new[] { typeof(string), typeof(string), typeof(StringComparison) }),
+                            propertyExpression, valueExpression, Expression.Constant(StringComparison.InvariantCultureIgnoreCase)),
+                        Expression.Constant(0));
+                case Condition.LessThanOrEqualTo:
+                    return Expression.LessThanOrEqual(Expression.Call(
+                            typeof(string).GetMethod("Compare", new[] { typeof(string), typeof(string), typeof(StringComparison) }),
+                            propertyExpression, valueExpression, Expression.Constant(StringComparison.InvariantCultureIgnoreCase)),
+                        Expression.Constant(0));
+                case Condition.IsNull:
+                    return Expression.Equal(propertyExpression, Expression.Constant(null));
+                case Condition.NotIsNull:
+                    return Expression.Not(Expression.Equal(propertyExpression, Expression.Constant(null)));
+                case Condition.IsEmpty:
+                    return Expression.Equal(propertyExpression, Expression.Constant(string.Empty));
+                case Condition.NotIsEmpty:
+                    return Expression.Not(Expression.Equal(propertyExpression, Expression.Constant(string.Empty)));
+                default:
+                    throw new NotSupportedException(string.Format(
+                        LanguageHelper.GetString("FilterBuilder_Exception_Message_ConditionIsNotSupported_Pattern"),
+                        condition));
+            }
+        }
+
+        private static Expression BuildIsNullExpression(ParameterExpression parameterExpression, string propertyName)
+        {
+            var type = Expression.Property(parameterExpression, propertyName).Type;
+            if (type.IsGenericType && type.GetGenericTypeDefinition() == typeof(Nullable<>))
```

Comment:
use IsGenericTypeEx() if possible to make sure we maximize cross-platform code (for eventual future usage).